connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -171,4 +172,11 @@ object ScalarUserDefinedFunction {
       nullable = true,
       deterministic = true)
   }
+
+  private[sql] def apply(function: AnyRef, returnType: DataType): ScalarUserDefinedFunction = {
+    ScalarUserDefinedFunction(
+      function = function,
+      inputEncoders = Seq.empty[AgnosticEncoder[_]],
```

Comment:
I'd be nice to have to have actual encoders here. We could try to use some of the stuff in `JavaTypeInference` for this.